### PR TITLE
Generate zilla.yaml from an AsyncAPI definition

### LINKED ISSUE #256: Generate `zilla.yaml` from `AsyncAPI` specification
Generate `zilla.yaml` from a corresponding `AsyncAPI` definition.

See the [AsyncAPI specification 3.0.0 (Pre-release)](https://www.asyncapi.com/docs/reference/specification/v3.0.0-next-major-spec.3)

Note: we may need additional information beyond `AsyncAPI` definition to be able to fully populate the generated `zilla.yaml`.

### PR DESCRIPTION
**Pull request by attilakreiner**
## Description

Generate zilla.yaml from a corresponding AsyncAPI definition.

Fixes #256

### CODE REVIEW

File: incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/asyncapi/model2/Resolvable.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.command.config.internal.asyncapi.model2;
```
Comment (jfallows) [inline incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/asyncapi/model2/Resolvable.java:15]: Perhaps `view` is a useful alternative name to `model2`?

File: incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/asyncapi/model2/Channel2.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.command.config.internal.asyncapi.model2;
+
+import java.util.Map;
+
+import io.aklivity.zilla.runtime.command.config.internal.asyncapi.model.Channel;
+
+public final class Channel2 extends Resolvable<Channel>
```
Comment (jfallows) [inline incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/asyncapi/model2/Channel2.java:21]: Suggest rename to `ChannelView`.

File: incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/asyncapi/model2/Server2.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ */
+package io.aklivity.zilla.runtime.command.config.internal.asyncapi.model2;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+import io.aklivity.zilla.runtime.command.config.internal.asyncapi.model.Server;
+
+public final class Server2
```
Comment (jfallows) [inline incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/asyncapi/model2/Server2.java:23]: Suggest rename to `ServerView`.

File: incubator/command-config/src/main/moditect/module-info.java [hunk 4]
```diff
@@ -27,6 +27,8 @@
 
     opens io.aklivity.zilla.runtime.command.config.internal.openapi.model;
 
+    opens io.aklivity.zilla.runtime.command.config.internal.asyncapi.model;
+
```
Comment (jfallows) [inline incubator/command-config/src/main/moditect/module-info.java:31]: This is needed to support reflection for the parsing, right?

File: incubator/command-config/src/test/resources/io/aklivity/zilla/runtime/command/config/internal/asyncapi/http/proxy/complete/asyncapi.yaml [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  messages:
+    item:
+      name: event
+      title: An event
+      correlationId:
+        $ref: "#/components/correlationIds/itemsCorrelationId"
+      headers:
+        type: object
+        properties:
+          Idempotency-Key:
```
Comment (jfallows) [inline incubator/command-config/src/test/resources/io/aklivity/zilla/runtime/command/config/internal/asyncapi/http/proxy/complete/asyncapi.yaml:74]: Is this intended to be mixed case?

File: incubator/command-config/src/test/resources/io/aklivity/zilla/runtime/command/config/internal/asyncapi/http/proxy/tls/zilla.yaml [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    - exit: tls_server0
+      when:
+      - port: 9090
+  tls_server0:
+    vault: server
+    type: tls
+    kind: server
+    options:
+      keys:
+      - "${{env.TLS_SERVER_KEYS}}"
```
Comment (jfallows) [inline incubator/command-config/src/test/resources/io/aklivity/zilla/runtime/command/config/internal/asyncapi/http/proxy/tls/zilla.yaml:19]: Perhaps this should be singular, `TLS_SERVER_KEY` as it is one key only in the environment variable?

**Review by jfallows — CHANGES_REQUESTED** [review]

File: incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/airline/ConfigGenerator.java [hunk 7]
```diff
@@ -14,7 +14,24 @@
...
+    {
+        for (String envVar : unquotedEnvVars)
+        {
+            yaml = yaml.replaceAll(
+                Pattern.quote(String.format("\"${{env.%s}}\"", envVar)),
+                String.format("\\${{env.%s}}", envVar)
+            );
+        }
+        return yaml;
+    }
```
[the remark this answers is not in the document]
Comment (attilakreiner) [reply incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/airline/ConfigGenerator.java:36]: done

File: incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/asyncapi/http/proxy/AsyncApiHttpProxyConfigGenerator.java [hunk 8]
```diff
@@ -0,0 +1,504 @@
...
+import io.aklivity.zilla.runtime.engine.config.GuardedConfigBuilder;
+import io.aklivity.zilla.runtime.engine.config.NamespaceConfig;
+import io.aklivity.zilla.runtime.engine.config.NamespaceConfigBuilder;
+import io.aklivity.zilla.runtime.engine.config.RouteConfigBuilder;
+import io.aklivity.zilla.runtime.guard.jwt.config.JwtOptionsConfig;
+import io.aklivity.zilla.runtime.vault.filesystem.config.FileSystemOptionsConfig;
+
+public class AsyncApiHttpProxyConfigGenerator implements ConfigGenerator
+{
+    private final InputStream inputStream;
```
[the remark this answers is not in the document]
Comment (attilakreiner) [reply incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/asyncapi/http/proxy/AsyncApiHttpProxyConfigGenerator.java]: done

File: incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/asyncapi/model2/Channel2.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+{
+    private final Channel channel;
+
+    private Channel2(
+        Map<String, Channel> channels,
+        Channel channel)
+    {
+        super.map = channels;
+        super.regex = "#/channels/(\\w+)";
```
[the remark this answers is not in the document]
Comment (attilakreiner) [reply incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/asyncapi/model2/Channel2.java]: done

File: incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/asyncapi/model2/Resolvable.java (hunk 1 above)
Comment (attilakreiner) [reply incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/asyncapi/model2/Resolvable.java]: Yeah, thanks, I like it, I updated it everywhere.

File: incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/asyncapi/model2/Channel2.java (hunk 2 above)
Comment (attilakreiner) [reply incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/asyncapi/model2/Channel2.java]: done

File: incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/asyncapi/model2/Server2.java (hunk 3 above)
Comment (attilakreiner) [reply incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/asyncapi/model2/Server2.java]: done

File: incubator/command-config/src/main/moditect/module-info.java (hunk 4 above)
Comment (attilakreiner) [reply incubator/command-config/src/main/moditect/module-info.java:31]: Yep, same situation as with the `*.openapi.model` that we discussed some time back... The json parser needs the access and I couldn't figure out how to narrow down the scope of `opens`.

File: incubator/command-config/src/test/resources/io/aklivity/zilla/runtime/command/config/internal/asyncapi/http/proxy/complete/asyncapi.yaml (hunk 5 above)
Comment (attilakreiner) [reply incubator/command-config/src/test/resources/io/aklivity/zilla/runtime/command/config/internal/asyncapi/http/proxy/complete/asyncapi.yaml]: Changed it to lowercase everywhere, probably a typo.

File: incubator/command-config/src/test/resources/io/aklivity/zilla/runtime/command/config/internal/asyncapi/http/proxy/tls/zilla.yaml (hunk 6 above)
Comment (attilakreiner) [reply incubator/command-config/src/test/resources/io/aklivity/zilla/runtime/command/config/internal/asyncapi/http/proxy/tls/zilla.yaml]: I changed it to singular everywhere. I am not sure why it was plural in the first place, probably just a typo.

**Review by jfallows — APPROVED** [review]

**Pull request merged**